### PR TITLE
role2

### PR DESCRIPTION
**Pull request by baaohuii**

### CODE REVIEW

**Comment by baaohuii** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -20,6 +26,8 @@ class User < ApplicationRecord
   before_save :downcase_email
   before_create :create_activation_digest
 
+  enum role: [:user0, :admin]
```
Comment (saobv-2223) [inline app/models/user.rb]:
```suggestion
  enum role: [user: 0, admin: 1]
```

File: app/models/user.rb [hunk 2]
```diff
@@ -1,6 +1,12 @@
 class User < ApplicationRecord
   attr_accessor :remember_token, :activation_token, :reset_token
 
+  after_initialize do
+    if self.new_record?
+      self.role ||= :user0
```
Comment (saobv-2223) [inline app/models/user.rb:6]: xem lại cách dùng enum nhé

**Comment by baaohuii** [comment]
ready

File: app/models/user.rb [hunk 3]
```diff
@@ -20,6 +26,8 @@ class User < ApplicationRecord
   before_save :downcase_email
   before_create :create_activation_digest
 
+  enum role: [user:0, admin:1]
```
Comment (ngocvt-0484) [inline app/models/user.rb]:
```suggestion
  enum role: {user: 0, admin: 1}
```

File: db/migrate/20201002005920_add_role_to_users.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class AddRoleToUsers < ActiveRecord::Migration[6.0]
+  def change
+    add_column :users, :role, :integer
```
Comment (ngocvt-0484) [inline db/migrate/20201002005920_add_role_to_users.rb]:
```suggestion
    add_column :users, :role, :integer, default: 0
```

Comment (ngocvt-0484) [reply db/migrate/20201002005920_add_role_to_users.rb]: before change you need run
```
rake db:rollback
```
and add default as above
and run migation again
```
rake db:migrate
```

File: app/models/user.rb [hunk 5]
```diff
@@ -1,6 +1,12 @@
 class User < ApplicationRecord
   attr_accessor :remember_token, :activation_token, :reset_token
 
+  after_initialize do
```
Comment (ngocvt-0484) [inline app/models/user.rb]: if you add migration as below, you not need this code
so please remove it

**Comment by baaohuii** [comment]
ready

**Pull request merged**